receiver/azureeventhubreceiver/azure_formatter_impl.go

```diff
@@ -0,0 +1,204 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azureeventhubreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azureeventhubreceiver"
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/relvacode/iso8601"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+const azureCategory = "azure.category"
+const azureCorrelationID = "azure.correlation.id"
+const azureDuration = "azure.duration"
+const azureIdentity = "azure.identity"
+const azureOperationName = "azure.operation.name"
+const azureOperationVersion = "azure.operation.version"
+const azureProperties = "azure.properties"
+const azureResourceID = "azure.resource.id"
+const azureResultType = "azure.result.type"
+const azureResultSignature = "azure.result.signature"
+const azureResultDescription = "azure.result.description"
+const azureTenantID = "azure.tenant.id"
```

Comment:
These look like they should be part of the `semantic convention`, is that right?

If they are not, it would be nice if they could be added to the project so that we can use the package `https://github.com/open-telemetry/opentelemetry-collector/tree/main/semconv/v1.9.0` to define them and allow translations.

Reply:
So it looks like the following need to be updated in the semantic conventions:
* [specification/logs/data-model.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md): Specifically Appendix A: Example Mappings.
* [cloud_provider/aws/logs.md](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/cloud_provider/aws/logs.md): Create a parallel file for Azure at `cloud_provider/azure/logs.md`
 
Happy to create a pull request for these attributes in the specification repository.

@atoulme @MovieStoreGuy Do you all want that to be reviewed and merged before merging this request? Or to create an issue to update this code once that request has been merged?